### ISSUE TITLE
[Truffle] Glob Issue needed for MRI tests

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#2825]
Sample Code (you'll need to just the path here to your test_alias.rb path):
```ruby
path = "/Users/brandonfish/Documents/jruby-mine/test/mri/ruby/test_alias.rb"
testfile_prefix = "test"
emb = "#{path}/**/#{testfile_prefix}_*.rb"
puts emb
puts Dir[emb].to_s
```
Expected:
```
/Users/brandonfish/Documents/jruby-mine/test/mri/ruby/test_alias.rb/**/test_*.rb
[]
```
Actual:
```
/Users/brandonfish/Documents/jruby-mine/test/mri/ruby/test_alias.rb/**/test_*.rb
["/Users/brandonfish/Documents/jruby-mine/test/mri/ruby/test_alias.rb"]
```

**Comment by bjfish** [comment]
The change in `unit.rb` where it says `if false` can be used to workaround this issue to run jt test mri tests:
https://gist.github.com/bjfish/1be84623da4a746f4d16

**Comment by chrisseaton** [comment]
I'm working on this at the moment - ended up pulling in a lot of new code to make it work and just fixing some bugs now.

**Comment by chrisseaton** [comment]
Fixed in 3063622f3c70ae50d859d0c45555aa6e2fb2f0b8

**Comment by bjfish** [comment]
@chrisseaton Re-opening because seeing the following error with the example:
```
/Users/brandonfish/Documents/jruby-mine/test/mri/ruby/test_alias.rb/**/test_*.rb
core:/core/rubinius/common/dir_glob.rb:107:in `const_missing': uninitialized constant Errno::ENOTDIR (NameError)
	from core:/core/rubinius/common/dir_glob.rb:107:in `call'
	from core:/core/rubinius/common/dir_glob.rb:63:in `call'
	from core:/core/rubinius/common/dir_glob.rb:84:in `call'
	from core:/core/rubinius/common/dir_glob.rb:324:in `run'
	from core:/core/rubinius/common/dir_glob.rb:402:in `glob'
	from core:/core/rubinius/common/dir.rb:64:in `block in glob'
	from core:/core/rubinius/common/dir.rb:61:in `each'
	from core:/core/rubinius/common/dir.rb:61:in `glob'
	from core:/core/rubinius/common/dir.rb:44:in `[]'
	from glob.rb:5:in `<main>'
```

**Comment by chrisseaton** [comment]
Ok - this should get resolved as part of my general overhaul of IO.

**Comment by chrisseaton** [comment]
Fixed in #2866
